go

```diff
@@ -95,9 +95,31 @@ def build
   exec_cmd "bundle #{JEKYLL_BUILD_CMD} #{JEKYLL_PUBLIC_CONFIG}"
 end
 
+
+PROOFER_OPTS = {
+  disable_external: true # might want to re-enable this eventually
+}
+
+def validate_public
+  require 'html/proofer'
+  HTML::Proofer.new('./_site_public', PROOFER_OPTS).run
+end
+
+def validate_private
+  require 'html/proofer'
+  HTML::Proofer.new('./_site', PROOFER_OPTS).run
+end
+
+def validate
+  build
+  validate_public
+  validate_private
+end
+
 def ci_build
   test
   build
+  validate_public
```

Comment:
Maybe we should take this out or rescue the exception so that we can merge this without all of our builds breaking until we fix them...